sw/airborne/arch/chibios/mcu_arch.c

```diff
@@ -54,6 +54,31 @@ static void init_ram_areas(const ram_init_area_t *rap);
 static void mpuConfigureNonCachedRam(void);
 #endif
 
+CH_IRQ_HANDLER(HardFault_Handler)
+{
+  mcu_reboot(MCU_REBOOT_FAST);
+}
+
+CH_IRQ_HANDLER(NMI_Handler)
+{
+  mcu_reboot(MCU_REBOOT_FAST);
+}
+
+CH_IRQ_HANDLER(MemManage_Handler)
+{
+  mcu_reboot(MCU_REBOOT_FAST);
+}
+
+CH_IRQ_HANDLER(BusFault_Handler)
+{
+  mcu_reboot(MCU_REBOOT_FAST);
+}
+
+CH_IRQ_HANDLER(UsageFault_Handler)
+{
+  mcu_reboot(MCU_REBOOT_FAST);
+}
+
```

Comment:
It seems to be colliding with the hard fault recovery mode. What is the purpose of this ?

Reply:
This needs some rewrite, but we don't want the hard fault recovery. But still reboot and skip the bootloader